src/Network.py

```diff
@@ -166,12 +166,25 @@ def connection_state(self) -> str:
     def load_show_file(self, xml: ET.Element, goto_default_scene: bool) -> None:
         msg = proto.FilterMode_pb2.load_show_file(show_data=ET.tostring(xml, encoding='utf8', method='xml'),
                                                   goto_default_scene=goto_default_scene)
-        self._send_with_format(msg, proto.MessageTypes_pb2.MSGT_LOAD_SHOW_FILE)
+        self._send_with_format(msg.SerializeToString(), proto.MessageTypes_pb2.MSGT_LOAD_SHOW_FILE)
 
     def enter_scene(self, scene_id: int) -> None:
         msg = proto.FilterMode_pb2.enter_scene(scene_id=scene_id)
         self._send_with_format(msg, proto.MessageTypes_pb2.MSGT_ENTER_SCENE)
 
+    def update_state(self, run_mode: proto.RealTimeControl_pb2.RunMode):
+        msg = proto.RealTimeControl_pb2.update_state(new_state=run_mode)
+        self._send_with_format(msg.SerializeToString(), proto.MessageTypes_pb2.MSGT_UPDATE_STATE)
+
+    def switch_to_direct(self):
+        self.update_state(proto.RealTimeControl_pb2.RunMode.RM_DIRECT)
+
+    def switch_to_filter(self):
+        self.update_state(proto.RealTimeControl_pb2.RunMode.RM_FILTER)
+
+    def switch_to_stop(self):
```

Comment:
I'd call this method stop_fish or something. `switch_to_stop` just sounds odd in my opinion.